contracts/lbp/LBPManager.sol

```diff
@@ -112,52 +126,60 @@ contract LBPManager {
     }
 
     /**
-     * @dev                             approve tokens for the vault and join pool to provide liquidity
-     * @param _tokens                   array of tokens sorted for the LBP
-     * @param _sender                   address who will send the tokens to add liquidity
-     * @param _fromInternalBalance      fund tokens from the internal user balance
-     * @param _userData                 userData specifies the type of join
+     * @dev                             Subtracts the primeDaoFee and adds liquidity to the LBP.
+     * @param _tokenList                Sorted array containing two parameters:
+                                            - The address of the project token being distributed.
+                                            - The address of the funding token being exchanged for the project token.
+     * @param _fundingTokenIndex        Index for the _tokenList array for the funding token.
+     * @param _sender                   Address of the liquidity provider.
+     * @param _userData                 UserData object that specifies the type of join.
      */
-    function fundPool(
-        IERC20[] memory _tokens,
+    function addLiquidity(
+        IERC20[] memory _tokenList,
+        uint8 _fundingTokenIndex,
         address _sender,
-        bool _fromInternalBalance,
         bytes memory _userData
-    ) public onlyAdmin {
+    ) external onlyAdmin {
         require(!poolFunded, "LBPManager: pool has already been funded");
         poolFunded = true;
 
         IVault vault = lbp.getVault();
 
         if (primeDaoFeePercentage != 0) {
-            // transfer primeDaoFee to beneficiary
-            _tokens[0].transferFrom(_sender, beneficiary, feeAmountRequired());
+            // Transfer primeDaoFee to beneficiary.
+            _tokenList[_fundingTokenIndex].transferFrom(
+                _sender,
+                beneficiary,
+                feeAmountRequired(_fundingTokenIndex)
+            );
         }
 
-        for (uint8 i; i < _tokens.length; i++) {
-            _tokens[i].transferFrom(_sender, address(this), amounts[i]);
-            _tokens[i].approve(address(vault), amounts[i]);
+        for (uint8 i; i < _tokenList.length; i++) {
+            _tokenList[i].transferFrom(_sender, address(this), amounts[i]);
+            _tokenList[i].approve(address(vault), amounts[i]);
         }
 
         IVault.JoinPoolRequest memory request = IVault.JoinPoolRequest({
             maxAmountsIn: amounts,
             userData: _userData,
-            fromInternalBalance: _fromInternalBalance,
-            assets: _tokens
+            fromInternalBalance: false, // It is not possible to add liquidity thought the internal Vault balance.
+            assets: _tokenList
```

Comment:
In light of what we talked about yesterday @Zitzak: this is under the assumption that we already receive the parameters sorted by address numerically from the frontend, right?